index.js

```diff
@@ -172,27 +165,173 @@ async function getSwapTovariableStats() {
+      if (tx.from === ACI_ETH && tx.to) {
+        return {
+          address: tx.to,
+          txHash: tx.hash,
+        };
+      } else {
+        return null;
+      }
+    });
+    addressesToCheck = addressesToCheck.filter((address) => address !== null); // remove nulls
+    addressesToCheck = addressesToCheck.filter(
+      (v, i, a) => a.findIndex((t) => t.address === v.address) === i
+    ); // remove duplicates
+
+    for (const user of addressesToCheck) {
+      const address = await isSafeWallet(user.address);
+      if (address) {
+        const gas = await getGasFromTx(user.txHash);
+        delegates[idx].safeWalletInteractions =
+          delegates[idx].safeWalletInteractions.add(gas);
+      }
     }
+  };
+
+  const addressesToCheck = [ACI_ETH, AAVECHAN_ETH];
+
+  for (const address of addressesToCheck) {
+    const idx = findDelegateIndex(address);
+    if (idx === -1) continue;
+
+    await safeWalletInteractions(address, idx);
   }
 }
 
+// Inspired from:
+// - https://github.com/abipub/evm-proxy-detection
+// - https://ethereum.stackexchange.com/a/141258
+const isSafeWallet = async (proxyAddress) => {
+  // found at https://github.com/safe-global/safe-deployments/tree/main/src/assets
+  const singletonsAddressesMainnet = [
+    "0x8942595a2dc5181df0465af0d7be08c8f23c93af", // v0.1.0 // found at https://help.safe.global/en/articles/40834-verify-safe-creation
+    "0xb6029EA3B2c51D09a50B53CA8012FeEB05bDa35A", // v1.0.0
+    "0x34CfAC646f301356fAa8B21e94227e3583Fe3F5F", // v1.1.0
+    "0x6851D6fDFAfD08c0295C392436245E5bc78B0185", // v1.2.0
+    "0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552", // v1.3.0 canonical
+    "0x69f4D1788e39c87893C980c06EdF4b7f686e2938", // v1.3.0 eip155
+    "0xB00ce5CCcdEf57e539ddcEd01DF43a13855d9910", // v1.3.0 zksync
+    "0x3E5c63644E683549055b9Be8653de26E0B4CD36E", // v1.3.0 L2 canonical
+    "0xfb1bffC9d739B8D520DaF37dF666da4C687191EA", // v1.3.0 L2 eip155
+    "0x1727c2c531cf966f902E5927b98490fDFb3b2b70", // v1.3.0 L2 zksync
+    "0x41675C099F32341bf84BFc5382aF534df5C7461a", // v1.4.1
+    "0x29fcB43b46531BcA003ddC8FCB67FFE91900C762", // v1.4.1 L2
+  ];
+
+  const readAddress = (value) => {
+    if (typeof value !== "string" || value === "0x") {
+      // throw new Error(`Invalid address value: ${value}`);
+      return null;
+    }
+
+    let address = value;
+    if (address.length === 66) {
+      address = "0x" + address.slice(-40);
+    }
+
+    if (address === constants.AddressZero) {
+      // throw new Error("Empty address");
+      return null;
+    }
+
+    if (
+      !singletonsAddressesMainnet.includes(ethers.utils.getAddress(address))
+    ) {
+      return null;
+    }
+
+    return address;
+  };
+
+  const SAFE_PROXY_INTERFACE = [
+    // bytes4(keccak256("masterCopy()")) padded to 32 bytes
+    "0xa619486e00000000000000000000000000000000000000000000000000000000",
+  ];
+
+  const infuraProvider = new InfuraProvider(1, process.env.INFURA_API_KEY);
```

Comment:
Please update example.env

Reply:
Ah my bad, I wasn't using INFURA_API_KEY on my side too ahah
I'll modify it to use jsonRpcProvider with defillama RPC url instead 👍